packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -53,50 +55,63 @@ module.exports = class RequestClient {
   }
 
   get (path) {
-    return fetch(this._getUrl(path), {
-      method: 'get',
-      headers: this.headers,
-      credentials: 'same-origin'
-    })
-      // @todo validate response status before calling json
-      .then(this.onReceiveResponse)
-      .then((res) => res.json())
-      .catch((err) => {
-        throw new Error(`Could not get ${this._getUrl(path)}. ${err}`)
+    return new Promise((resolve, reject) => {
```

Comment:
This promise wrapper should be unnecessary—can `return` the `this.headers().then()` chain instead

Reply:
yes, but I just generally find Promise chaining hard to follow. It makes it hard to tell what value eventually gets resolved. But maybe I just need to get used to the syntax

Reply:
I find the nesting harder to follow :smile: I prefer flatter chains quite a lot, but if it's clearer to you this way it's OK.